StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1024CodeFixProvider.cs

```diff
@@ -0,0 +1,131 @@
+
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                if (!diagnostic.Id.Equals(SA1024ColonsMustBeSpacedCorrectly.DiagnosticId))
+                {
+                    continue;
+                }
+
+                SyntaxToken token = root.FindToken(diagnostic.Location.SourceSpan.Start);
+                if (token.IsMissing)
+                {
+                    continue;
+                }
+
+                context.RegisterCodeFix(CodeAction.Create(SpacingResources.SA1024CodeFix, t => GetTransformedDocumentAsync(context.Document, root, token)), diagnostic);
+            }
+        }
+
+        private static Task<Document> GetTransformedDocumentAsync(Document document, SyntaxNode root, SyntaxToken token)
+        {
+            bool requireBefore = true;
+            switch (token.Parent.Kind())
+            {
+                case SyntaxKind.LabeledStatement:
+                case SyntaxKind.CaseSwitchLabel:
+                case SyntaxKind.DefaultSwitchLabel:
+                // NameColon is not explicitly listed in the description of this warning, but the behavior is inferred
+                case SyntaxKind.NameColon:
+                    requireBefore = false;
+                    break;
+            }
+
+            // check for a following space
+            bool missingFollowingSpace = true;
+            if (token.HasTrailingTrivia)
+            {
+                if (token.TrailingTrivia.First().IsKind(SyntaxKind.WhitespaceTrivia))
+                {
+                    missingFollowingSpace = false;
+                }
+                else if (token.TrailingTrivia.First().IsKind(SyntaxKind.EndOfLineTrivia))
+                {
+                    missingFollowingSpace = false;
+                }
+            }
+
+            bool hasPrecedingSpace = token.HasLeadingTrivia;
+            if (!hasPrecedingSpace)
+            {
+                // only the first token on the line has leading trivia, and those are ignored
+                SyntaxToken precedingToken = token.GetPreviousToken();
+                SyntaxTriviaList triviaList = precedingToken.TrailingTrivia;
+                if (triviaList.Count > 0 && !triviaList.Last().IsKind(SyntaxKind.MultiLineCommentTrivia))
+                {
+                    hasPrecedingSpace = true;
+                }
+            }
+
+            if (missingFollowingSpace && requireBefore && !hasPrecedingSpace)
+            {
+                SyntaxTrivia whitespace = SyntaxFactory.Space;
+                SyntaxToken corrected = token.WithTrailingTrivia(token.TrailingTrivia.Insert(0, whitespace))
+                                             .WithLeadingTrivia(token.LeadingTrivia.Add(whitespace));
+                Document updatedDocument = document.WithSyntaxRoot(root.ReplaceToken(token, corrected));
+                return Task.FromResult(updatedDocument);
+            }
+            else if (missingFollowingSpace)
```

Comment:
The `else if` construction is not needed, as the successful `if` above will exit the method.